src/mono/wasm/Wasm.Build.Tests/WasmTemplateTestBase.cs

```diff
@@ -59,31 +59,55 @@ public string CreateWasmTemplateProject(string id, string template = "wasmbrowse
 
     public (string projectDir, string buildOutput) BuildTemplateProject(BuildArgs buildArgs,
         string id,
-        BuildProjectOptions buildProjectOptions,
-        AssertTestMainJsAppBundleOptions? assertAppBundleOptions = null)
+        BuildProjectOptions buildProjectOptions)
     {
         (CommandResult res, string logFilePath) = BuildProjectWithoutAssert(id, buildArgs.Config, buildProjectOptions);
         if (buildProjectOptions.UseCache)
             _buildContext.CacheBuild(buildArgs, new BuildProduct(_projectDir!, logFilePath, true, res.Output));
 
         if (buildProjectOptions.AssertAppBundle)
-            AssertBundle(buildArgs, buildProjectOptions, res.Output, assertAppBundleOptions);
+        {
+            if (buildProjectOptions.IsBrowserProject)
+                AssertWasmSdkBundle(buildArgs, buildProjectOptions, res.Output);
+            else
+                AssertTestMainJsBundle(buildArgs, buildProjectOptions, res.Output);
+        }
         return (_projectDir!, res.Output);
     }
 
-    public void AssertBundle(BuildArgs buildArgs,
+    public void AssertTestMainJsBundle(BuildArgs buildArgs,
```

Comment:
Keep the `AssertBundle`, but add an overload for the `WasmSdk` options.

Reply:
I can't do that because I'm calling these methods without the distinguishing parameter. And I can't create the parameter, because the creation of `Assert*BundleOptions` is in the provider. Having a separately named methods seems like an easiest win